library/src/main/java/com/crossbowffs/remotepreferences/RemotePreferenceProvider.java

```diff
@@ -76,20 +76,82 @@
      * authority and preference files. The authority must match the
      * {@code android:authorities} property defined in your manifest
      * file. Only the specified preference files will be accessible
-     * through the provider.
+     * through the provider. Kept for backward compatibility.
      *
      * @param authority The authority of the provider.
      * @param prefNames The names of the preference files to expose.
      */
     public RemotePreferenceProvider(String authority, String[] prefNames) {
+        this(authority, RemotePreferenceFile.fromStringArray(prefNames));
+    }
+
+    /**
+     * Another constructor for the provider.
+     *
+     * @param authority The authority of the provider.
+     * @param prefNames An array of the file names with their Contexts to expose.
+     *                  @see RemotePreferenceFile
+     */
+    public RemotePreferenceProvider(String authority, RemotePreferenceFile[] prefNames) {
         mBaseUri = Uri.parse("content://" + authority);
-        mPrefNames = prefNames;
+        mRemotePreferenceFiles = prefNames;
         mPreferences = new HashMap<String, SharedPreferences>(prefNames.length);
         mUriMatcher = new UriMatcher(UriMatcher.NO_MATCH);
         mUriMatcher.addURI(authority, "*/", PREFERENCES_ID);
         mUriMatcher.addURI(authority, "*/*", PREFERENCE_ID);
     }
 
+    /**
+     * Class for tagging each preference file's Context.
+     */
+    public static class RemotePreferenceFile {
+        /**
+         * The name of the preference file.
+         */
+        String prefFileName;
+        /**
+         * Context of the preference file.
+         * Use true if these preferences need to be exposed before the first unlock
+         * Or false, which is the default, if they are stored encrypted on the device.
+         */
+        boolean isDeviceProtected = false;
+
+        /**
+         * Initialize the object.
+         * @param name  Name of the preference file.
+         * @param state A boolean indicating whether the Context is device protected (true), or
+         *              credential protected (false).
+         */
+        public RemotePreferenceFile(String name, boolean state) {
+            prefFileName = name;
+            isDeviceProtected = state;
+        }
+
+        /**
+         * Another constructor for the object.
+         * Uses the credential protected context by default.
+         * @param name  Name of the preference file.
+         */
+        RemotePreferenceFile(String name) {
```

Comment:
Please explicitly add the accessibility modifiers here (I think `public` would be fine?)